controllers/helmchart_controller_test.go

```diff
@@ -183,12 +319,20 @@ func TestHelmChartReconciler_reconcileStorage(t *testing.T) {
 						Path:     fmt.Sprintf("/reconcile-storage/%s.txt", v),
 						Revision: v,
 					}
+					provArtifact := &sourcev1.Artifact{
+						Path:     fmt.Sprintf("/reconcile-storage/%s.txt.prov", v),
+						Revision: v,
+					}
+
 					if err := testStorage.MkdirAll(*obj.Status.Artifact); err != nil {
 						return err
 					}
 					if err := testStorage.AtomicWriteFile(obj.Status.Artifact, strings.NewReader(v), 0644); err != nil {
 						return err
 					}
+					if err := testStorage.AtomicWriteFile(provArtifact, strings.NewReader(v), 0644); err != nil {
```

Comment:
Let's use octal format for representing os-level octal values:
```suggestion
					if err := testStorage.AtomicWriteFile(provArtifact, strings.NewReader(v), 0o644); err != nil {
```

Reply:
Rather than making a couple of changes in this PR, I think this change should be addressed overall in the PR regarding #603

Reply:
I do agree on this for existing code, but this line specifically is a new introduction.